### ISSUE TITLE
selector-type-no-unknown regexp rules doesn't work

### ISSUE THREAD

**Issue by sormy** [stylelint/stylelint#1733]
A bug with `selector-type-no-unknown` rule, option `ignoreTypes` based on docs accepts regexps but it doesn't work. It accepts only strings, so it is either bug in documentation or not implemented functionality in the code.

example css:

```css
app-my-super-component {
  color: pink;
}
```

example configuration:

```json
{
  "rules": {
    "selector-type-no-unknown": [true, {
      "ignoreTypes": ["/^app-/"]
    }]
  }
}
```

stylelint: v7.0.3

How are you running stylelint:  vs code plugin with .stylelintrc in the project root

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no

> What did you expect to happen?

it should use regexp rule and do not show error message

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
stylelint: Unexpected unknown type selector "app-my-super-component" (selector-type-no-unknown)
```

**Comment by evilebottnawi** [comment]
@sormy please format you issue as template

**Comment by evilebottnawi** [comment]
@sormy for disable rules use `null`, example `"selector-type-no-unknown": null`

**Comment by sormy** [comment]
null works, thanks, may be that will be good to note in documentation

**Comment by evilebottnawi** [comment]
@sormy this note exist in docs https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#rules

**Comment by evilebottnawi** [comment]
@sormy can i close issue?

**Comment by sormy** [comment]
RegExp rules doesn't work anyway

**Comment by evilebottnawi** [comment]
@sormy format issue as template with example configuration and code, it will simplify the task to help you

**Comment by evilebottnawi** [comment]
@sormy Thanks for feedback, confirmed.
Ref: https://github.com/stylelint/stylelint/blob/master/src/rules/selector-type-no-unknown/index.js#L82

**Comment by evilebottnawi** [comment]
@sormy oh sorry, we already fixed it, i next release in 7.1.0 it is work fine

**Comment by evilebottnawi** [comment]
@jeddy3 why we need this line https://github.com/stylelint/stylelint/blob/master/src/rules/selector-type-no-unknown/index.js#L82 if above we have https://github.com/stylelint/stylelint/blob/master/src/rules/selector-type-no-unknown/index.js#L70

**Comment by jeddy3** [comment]
> RegExp rules doesn't work anyway

@sormy As @evilebottnawi mentioned, the regex support for `ignoreTypes` [is coming](https://github.com/stylelint/stylelint/blob/master/CHANGELOG.md#head) in the next release. For future reference, [the website](http://stylelint.io/) docs are reflective of what the *latest release* is, whereas the [repo docs](https://github.com/stylelint/stylelint/tree/master/docs) are reflective of the current state of *master*.
